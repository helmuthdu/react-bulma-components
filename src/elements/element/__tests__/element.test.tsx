import { render } from '@testing-library/react';
import PropTypes from 'prop-types';
import * as React from 'react';
import { Element } from '..';

describe('Element component', () => {
  it('should render', () => {
    const { asFragment } = render(
      // @ts-ignore
      <Element pull="left" textColor="white">
        Facebook
      </Element>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render component with different tag', () => {
    const Custom = (props: any) => (
      <p {...props}>
        Custom
        {props.children}
      </p>
    );

    Custom.propTypes = { children: PropTypes.node.isRequired };

    const { asFragment } = render(<Element as={Custom}>This should be a p element</Element>);
    expect(asFragment()).toMatchSnapshot();
  });
});
