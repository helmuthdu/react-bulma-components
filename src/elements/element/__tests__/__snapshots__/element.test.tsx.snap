// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Element component should render 1`] = `
<DocumentFragment>
  <div
    class="has-text-white"
    pull="left"
  >
    Facebook
  </div>
</DocumentFragment>
`;

exports[`Element component should render component with different tag 1`] = `
<DocumentFragment>
  <p>
    CustomThis should be a p element
  </p>
</DocumentFragment>
`;
